File: python/meteroid/models/credit_note_event_data.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .credit_note_id import CreditNoteId
from .credit_note_status import CreditNoteStatus
from .customer_id import CustomerId
from .invoice_id import InvoiceId
from .invoice_line_item import InvoiceLineItem
from .tax_breakdown_item import TaxBreakdownItem


@dataclasses.dataclass
class CreditNoteEventData(BaseModel):
    created_at: datetime

    credit_note_id: CreditNoteId

    credited_amount_cents: int

    currency: str

    custom_properties: t.Any
    """User-defined custom property values, keyed by definition key."""

    customer_id: CustomerId

    invoice_id: InvoiceId

    line_items: t.List[InvoiceLineItem]
    """Credited line items (negated amounts)."""

    refunded_amount_cents: int

    status: CreditNoteStatus

    subtotal: int

    tax_amount: int

    tax_breakdown: t.List[TaxBreakdownItem]
    """Per-rate tax (VAT) breakdown for the credited amount."""

    total: int

    credit_note_number: t.Optional[str] = None
    """Absent while the credit note is a draft — the number is assigned at finalization."""

    invoice_number: t.Optional[str] = None
    """Number of the invoice being credited."""

    memo: t.Optional[str] = None

    reason: t.Optional[str] = None

File: go/credit_note_event_data.go
// this file is @generated
package meteroid

import (
	"encoding/json"
	"time"
)

type CreditNoteEventData struct {
	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	CreditNoteId CreditNoteId `json:"credit_note_id"`

	// Absent while the credit note is a draft — the number is assigned at finalization.
	CreditNoteNumber *string `json:"credit_note_number,omitempty"`

	CreditedAmountCents int64 `json:"credited_amount_cents"`

	Currency string `json:"currency"`

	// User-defined custom property values, keyed by definition key.
	CustomProperties json.RawMessage `json:"custom_properties"`

	CustomerId CustomerId `json:"customer_id"`

	InvoiceId InvoiceId `json:"invoice_id"`

	// Number of the invoice being credited.
	InvoiceNumber *string `json:"invoice_number,omitempty"`

	// Credited line items (negated amounts).
	LineItems RequiredSlice[InvoiceLineItem] `json:"line_items"`

	Memo *string `json:"memo,omitempty"`

	Reason *string `json:"reason,omitempty"`

	RefundedAmountCents int64 `json:"refunded_amount_cents"`

	Status CreditNoteStatus `json:"status"`

	Subtotal int64 `json:"subtotal"`

	TaxAmount int64 `json:"tax_amount"`

	// Per-rate tax (VAT) breakdown for the credited amount.
	TaxBreakdown RequiredSlice[TaxBreakdownItem] `json:"tax_breakdown"`

	Total int64 `json:"total"`
}
